pytorch_tabnet/tab_network.py

```diff
@@ -267,13 +268,7 @@ def __init__(self, input_dim, output_dim, n_d=8, n_a=8,
                                          gamma, n_independent, n_shared, epsilon,
                                          virtual_batch_size, momentum, mask_type)
 
-        # Defining device
```

Comment:
Yes this is definitely not something we want to keep.
But do we want to call `define_device` in the `tab_network.py`?

Does something go wrong if we only define the device in the abstract model? and make sure to call the network with correct device?

Reply:
You mean in the `_set_network` function? I guess we could do that as well, it's probably cleaner.

Reply:
either in `_set_network` or `__post_init__`, but I think we should define this here once in `abstract_model.py` and not in `tab_network.py`

Reply:
actually I think `TabNet` class should not have a `device_name` parameter but just `device` and the abstract_model should define correctly what to use in _set_network and/or __post_init__